server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -242,34 +253,23 @@ private static CompletionStage<Long> counterIncOrDec(Cache<byte[], byte[]> cache
             });
    }
 
-   private CompletionStage<RespRequestHandler> performSet(ChannelHandlerContext ctx, Cache<byte[], byte[]> cache, byte[] key, byte[] value,
-         long lifespan, String type, ByteBuf messageOnSuccess) {
-      return stageToReturn(cache.putAsync(key, value, lifespan, TimeUnit.SECONDS), ctx, (ignore, t) -> {
-         if (t != null) {
-            log.trace("Exception encountered while performing " + type, t);
-            handleThrowable(ctx, t);
-         } else {
-            ctx.writeAndFlush(messageOnSuccess, ctx.voidPromise());
-         }
-      });
-   }
 
    private CompletionStage<RespRequestHandler> performDelete(ChannelHandlerContext ctx, Cache<byte[], byte[]> cache, List<byte[]> arguments) {
       int keysToRemove = arguments.size();
       if (keysToRemove == 1) {
          byte[] keyBytes = arguments.get(0);
-         return stageToReturn(cache.removeAsync(keyBytes), ctx, (prev, t) -> {
+         return stageToReturn(cache.removeAsync(keyBytes), ctx, (prev, innerCtx, t) -> {
```

Comment:
Actually, this should use a REMOVE_TRICONSUMER to avoid the allocation per remove command.